src/Compilers/CSharp/Portable/Lowering/AsyncRewriter/RuntimeAsyncRewriter.cs

```diff
@@ -161,4 +198,102 @@ public override BoundNode VisitAwaitableValuePlaceholder(BoundAwaitableValuePlac
     {
         return _placeholderMap[node];
     }
+
+    public override BoundNode? VisitAssignmentOperator(BoundAssignmentOperator node)
+    {
+        if (node.Left is not BoundLocal leftLocal)
+        {
+            return base.VisitAssignmentOperator(node);
+        }
+
+        BoundExpression originalRight = node.Right;
+        BoundExpression visitedRight;
+
+        if (_variablesToHoist.Contains(leftLocal.LocalSymbol) && !_proxies.ContainsKey(leftLocal.LocalSymbol))
+        {
+            visitedRight = VisitExpression(node.Right);
+            return _refInitializationHoister.HoistRefInitialization(
```

Comment:
At the other call-site for `HoistRefInitialization` we have a number of assertions, including about the type of local we should encounter and the assignment being by-reference. Can we do the same here?